gson/src/main/java/com/google/gson/stream/MutableJsonReader.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2020 https://github.com/911992.
```

Comment:
I am not sure if that is allowed, it might have to be (though I am not sure):
> Copyright 2020 Google Inc.

Reply:
Much appreciated for your review.

Please accept my apology, about mistakes I made.

Personally, I had the changes based on my needs, then I decided to contribute the opensource, that maybe could be handy for devs may work on high-demand/heavy enterprise env. As my tests, pooling indeed help to have a stable env with less possible/common memory related issues.

About the rules and the changes that must be applied, I understand, and I would change anything should be changed.

I'm going to apply the copyright notice too, sorry it's a automated header is made by my IDE. Sorry I didn't realize it.

Reply:
No problem. Though I am not a maintainer of this project and they might have different requirements for the pull request.